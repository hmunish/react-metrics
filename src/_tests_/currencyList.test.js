import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import CurrencyList from "../components/currencies/currencyList";

const mockStore = configureStore([]);

describe("test currency list component", () => {
  it("should render correct snapshot", () => {
    const mockCurrencies = {
      currenciesData: [],
      curCountryCode: "",
      isLoading: false,
      isError: false,
    };

    const store = mockStore({
      currency: mockCurrencies,
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <CurrencyList />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
