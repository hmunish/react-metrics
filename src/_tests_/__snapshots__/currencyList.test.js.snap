// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render currency items correctly should render error message 1`] = `
<p
  className="status"
  data-testid="error-msg"
/>
`;

exports[`test currency list component should render correct snapshot 1`] = `
<section
  className="currencies"
>
  <div
    className="overview"
  >
    <div />
    <h3>
      World currencies names
    </h3>
  </div>
  <h1
    className="title"
  >
    Stats by currency
  </h1>
  <ul
    className="countryList"
  />
</section>
`;
